### PR TITLE
Modified JDAO to continue parsing lines through parse errors

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Modified JDAO to not stop on parse failure. 
2. Added different parsing of lines depending on the operation

### CODE REVIEW

File: src/foam/dao/AbstractJDAO.java [hunk 1]
```diff
@@ -79,18 +79,19 @@ protected void loadJournal()
         // remove first two characters and last character
         line = line.trim().substring(2, line.trim().length() - 1);
 
-        FObject object = parser.parseString(line);
-        if ( object == null ) {
-          System.err.println(getParsingErrorMessage(line) + ", source: " + line);
-          break;
```
Comment (kirkeaton) [inline src/foam/dao/AbstractJDAO.java]: @kgrgreer @jlhughes This break statement was causing a journal file to stop being parsed

**Review by jlhughes — COMMENTED** [review]
Are parsing failures caught in the 'catch'? Otherwise can you log them when object is nil?

**Comment by kirkeaton** [comment]
@jlhughes I don't think they will be caught in the catch but I will log them.

**Review by jlhughes — APPROVED** [review]

File: src/foam/dao/AbstractJDAO.java [hunk 2]
```diff
@@ -79,20 +79,26 @@ protected void loadJournal()
...
           case 'p':
-            getDelegate().put(object);
+            object = parser.parseString(line);
+            if ( object != null ) {
+              getDelegate().put(object);
+            }
             break;
           case 'r':
-            getDelegate().remove(object);
+            Object id = parser.parseStringForId(line);
```
Comment (adamvy) [inline src/foam/dao/AbstractJDAO.java]: There's a reason I removed the parseStringForId line and used parseObject in both cases.

parseStringForId fails for multipart ID objects.

Comment (kirkeaton) [reply src/foam/dao/AbstractJDAO.java]: Okay then how do you propose we solve this. The current remove lines that are printed out look like this: ```r({"id":1})``` and parseString does not parse this either

Comment (adamvy) [reply src/foam/dao/AbstractJDAO.java]: I changed the remove lines to output the full object as well when I made this change.  I propose leaving it as is.

Comment (adamvy) [reply src/foam/dao/AbstractJDAO.java]: https://github.com/foam-framework/foam2/commit/a501a9c7b6544154aa9aadae7aa93578c43a495a

Comment (kirkeaton) [reply src/foam/dao/AbstractJDAO.java]: Actually I see that you modified the remove_ function to print out the whole object just like the put_ method. But as it stands the old remove lines are broken.

Comment (adamvy) [reply src/foam/dao/AbstractJDAO.java]: Do we have existing data in the old format we need to support?

I guess we could try to parse it both ways.  If the current method fails to parse, then try to parse it as just an id.

Comment (kirkeaton) [reply src/foam/dao/AbstractJDAO.java]: @adamvy Okay leaving it as is but instead of stopping on parse failure Ill change it to just continue.

**Review by adamvy — APPROVED** [review]

**Pull request merged**